### PR TITLE
feat(locale): add directions and directions abbr to pl

### PR DESCRIPTION
**Pull request by hankucz**

### CODE REVIEW

File: src/locales/pl/address/direction.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default [
+  'Północ',
```
Comment (pkuczynski) [inline src/locales/pl/address/direction.ts:2]: * add source: https://pl.wikipedia.org/wiki/Strony_świata
* sort alphabetically
* use lower case

**Review by import-brain — APPROVED** [review]
Translations verified using Google Translate, lgtm

File: src/locales/pl/address/direction_abbr.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export default ['N', 'E', 'S', 'W', 'NE', 'NW', 'SE', 'SW'];
```
Comment (pkuczynski) [inline src/locales/pl/address/direction_abbr.ts:1]: There are polish abbreviations: pn., pd., ...

See: https://pl.wikipedia.org/wiki/Strony_świata

**Review by pkuczynski — CHANGES_REQUESTED** [review]

File: src/locales/pl/address/direction_abbr.ts (hunk 2 above)
Comment (hankucz) [reply src/locales/pl/address/direction_abbr.ts]: Done...

**Review by pkuczynski — DISMISSED** [review]

File: src/locales/pl/address/direction.ts (hunk 1 above)
Comment (hankucz) [reply src/locales/pl/address/direction.ts]: Done...

**Review by import-brain — DISMISSED** [review]

**Review by ST-DDT — CHANGES_REQUESTED** [review]
The order of the elements is somewhat important:
https://github.com/faker-js/faker/blob/04c9d36cc44fa647f401919c708e550dbe42040d/src/modules/address/index.ts#L411-L453
0-3 => cardinal direction
4-7 => ordinal direction

**Comment by pkuczynski** [comment]
Wow! I would never expect this and it's really hard to catch on the review. It would be way better if TS could catch it automatically...

Should we change the definition to look more like:

```js
interface Directions {
  cardinal: string[]
  ordinal: string[] // or intercardinal ?
  secondaryIntercardinal?: string[]
}

type DirectionType = keyof Directions
```

And then both `directions` and `directions_abbr` should use this type?

Additionally, I think it makes sense to simplify the usage and change `direction` signature to:

```
direction ({ abbr: boolean, type: DirectionType, types: DirectionType[]})

@examples
faker.address.direction() // 'north'
faker.address.direction({ abbr: true }) // 'N'
faker.address.direction({ type: 'ordinal' }) // 'north west'
faker.address.direction({ abbr: true, type: 'ordinal' }) // 'NW'
faker.address.direction({ abbr: true, types: ['ordinal', 'secondaryIntercardinal']  }) // 'NNW'
```

And then deprecate the other methods? We might also skip `secondaryIntercardinal`...

**Comment by ST-DDT** [comment]
@pkuczynski Please create a separate issue for that.

**Comment by pkuczynski** [comment]
> @pkuczynski Please create a separate issue for that.

Sure! https://github.com/faker-js/faker/issues/1228

**Comment by ST-DDT** [comment]
@hankucz Could you please restore the original order? (N,E,S,W, NE, SE, SW, SE)
That way we can merge this and can easily automatically transform it once we change the implementation.

**Comment by hankucz** [comment]
@ST-DDT original order restored

**Review by pkuczynski — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**